traits/trait_numeric.py

```diff
@@ -296,104 +283,83 @@ def _default_for_dtype_and_shape(self, dtype, shape):
+        ranges are allowed. The value None within the *shape* tuple means
+        that the corresponding dimension is not checked. (For example,
+        ``shape=(None,3)`` means that the first dimension can be any size,
+        but the second must be 3.) A two-element tuple within the *shape*
+        tuple means that the dimension must be in the specified range. The
+        second element can be None to indicate that there is no upper
+        bound. (For example, ``shape=((3,5),(2,None))`` means that the
+        first dimension must be in the range 3 to 5 (inclusive), and the
+        second dimension must be at least 2.)
+    value : numpy array
+        A default value for the array.
     """
 
     def __init__(
         self, dtype=None, shape=None, value=None, typecode=None, **metadata
     ):
-        """ Returns an Array trait.
-
-        Parameters
-        ----------
-        dtype : a numpy dtype (e.g., int32)
-            The type of elements in the array; if omitted, no type-checking is
-            performed on assigned values.
-        shape : a tuple
-            Describes the required shape of any assigned value. Wildcards and
-            ranges are allowed. The value None within the *shape* tuple means
-            that the corresponding dimension is not checked. (For example,
-            ``shape=(None,3)`` means that the first dimension can be any size,
-            but the second must be 3.) A two-element tuple within the *shape*
-            tuple means that the dimension must be in the specified range. The
-            second element can be None to indicate that there is no upper
-            bound. (For example, ``shape=((3,5),(2,None))`` means that the
-            first dimension must be in the range 3 to 5 (inclusive), and the
-            second dimension must be at least 2.)
-        value : numpy array
-            A default value for the array.
-
-        Default Value
-        -------------
-        *value* or ``zeros(min(shape))``, where ``min(shape)`` refers to the
-        minimum shape allowed by the array. If *shape* is not specified, the
-        minimum shape is (0,).
-
-        Description
-        -----------
-        An Array trait allows only upcasting of assigned values that are
-        already numpy arrays. It automatically casts tuples and lists of the
-        right shape to the specified *dtype* (just like numpy's **array**
-        does).
-        """
         super(Array, self).__init__(
             dtype, shape, value, False, typecode=typecode, **metadata
         )
 
 
-# -------------------------------------------------------------------------------
-#  'CArray' trait:
-# -------------------------------------------------------------------------------
-
-
 class CArray(AbstractArray):
-    """ Defines a trait whose value must be a numpy array, with casting
-        allowed.
+    """ A coercing trait type whose value is a numpy array.
```

Comment:
Just for consistency:

```suggestion
    """ A coercing trait type whose value is a NumPy array.
```